File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
    
        if(s.length()!= t.length()) return false;
        Map<Character,Integer> map = new HashMap<>();
        for(char x : s.toCharArray())
        {
           map.put(x, map.getOrDefault(x, 0) + 1);
        }
        for(char z : t.toCharArray())
        {
            map.put(z, map.getOrDefault(z,0)-1);
        }
        for(int value:map.values())
        {
            if (value!=0) return false;

        }
        return true;
    




    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashmapS = dict()
        hashmapT = dict()
        if len(s) != len(t):
            return False
        for char in s:
            hashmapS[char] = hashmapS.get(char,0) +1
        for char in t:
            hashmapT[char] = hashmapT.get(char,0) + 1
        for char in s:
            if hashmapT.get(char,0) != hashmapS.get(char,0):
                return False
        return True


        